DesktopEdge/Views/Screens/IdentityDetails.xaml.cs

```diff
@@ -217,6 +218,36 @@ public partial class IdentityDetails:UserControl {
 			DetailPorts.Text = info.PortString;
 			DetailUrl.Text = info.ToString();
 
+			try {
+				if (_identity.IsMFAEnabled) {
+					if (info.TimeoutRemaining > 0) {
+						// t = TimeSpan.FromSeconds(info.TimeoutRemaining);
+						TimeSpan t = (DateTime.Now - info.TimeUpdated);
+						int timeout = info.Timeout - (int)Math.Floor(t.TotalSeconds);
+
+						if  (timeout>0) {
+							t = TimeSpan.FromSeconds(timeout);
+							string answer = string.Format("{0:D2} minutes", t.Minutes);
+							if (t.Days > 0) answer = string.Format("{0:D2} days {1:D2} hours {2:D2} minutes", t.Days, t.Hours, t.Minutes);
+							else {
+								if (t.Hours > 0) answer = string.Format("{0:D2} hours {1:D2} minutes", t.Hours, t.Minutes);
+							}
+							TimeoutDetails.Text = answer;
+						} else {
+							TimeoutDetails.Text = "Timed Out";
+						}
+
+					} else {

```

Comment:
If TimeoutRemaining == -1, it means it never times out.
If TimeoutRemaining == 0, it means it has timed out.